qa/logging-config/src/test/java/org/elasticsearch/common/logging/JsonLoggerTests.java

```diff
@@ -101,9 +103,130 @@ public void testDeprecatedMessage() throws IOException {
         }
     }
 
+    public void testMessageOverrideWithNoValue() throws IOException {
+        //message field is meant to be overriden (see custom.test config), but is not provided.
+        //Expected is that it will be emptied
+        final Logger testLogger = LogManager.getLogger("custom.test");
+
+        testLogger.info(ESLogMessage.message("some message"));
+
+        final Path path = PathUtils.get(System.getProperty("es.logs.base_path"),
+            System.getProperty("es.logs.cluster_name") + "_custom.json");
+        try (Stream<Map<String, String>> stream = JsonLogsStream.mapStreamFrom(path)) {
+            List<Map<String, String>> jsonLogs = stream
+                .collect(Collectors.toList());
+
+            assertThat(jsonLogs, contains(
+                allOf(
+                    hasEntry("type", "custom"),
+                    hasEntry("level", "INFO"),
+                    hasEntry("component", "c.test"),
+                    hasEntry("cluster.name", "elasticsearch"),
+                    hasEntry("node.name", "sample-name"))
+                )
+            );
+        }
+    }
+    public void testBuildingMessage() throws IOException {
```

Comment:
@atorok how about this approach for building messages? 

This is influenced by @felixbarny [ecs testcase](https://github.com/elastic/java-ecs-logging/blob/master/log4j2-ecs-layout/src/test/java/co/elastic/logging/log4j2/ParameterizedStructuredMessage.java)
Should work out of the box if we migrate to ECS

Reply:
Thanks, I think this looks much better